packages/graphql-generator/src/utils/GraphQLStatementsFormatter.ts

```diff
@@ -36,10 +36,9 @@ export class GraphQLStatementsFormatter {
     this.lintOverrides = [];
     this.headerComments = [];
     if (typesPath) {
+      const { dir, name } = path.parse(typesPath);
       // ensure posix path separators are used
-      const typesPathWithPosixSep = typesPath.split(path.win32.sep).join(path.posix.sep)
-      const { dir, name } = path.parse(typesPathWithPosixSep);
-      const typesPathWithoutExtension = path.join(dir, name);
+      const typesPathWithoutExtension = path.join(dir, name).split(path.win32.sep).join(path.posix.sep);
```

Comment:
Fixing the issue introduced by https://github.com/aws-amplify/amplify-codegen/pull/742. Shows how easy it is to mess up without the unit testing in Windows. Thankfully the change has not been released.